### PR TITLE
Rspec test for Tic Tac Toe game

### PR DESCRIPTION
**Pull request by brytebee**
This branch seeks to implement Test-Driven Development.
It seeks to test the methods in the Tic Tac Toe game

### CODE REVIEW

**Review by george-shammar — CHANGES_REQUESTED** [review]
### **STATUS: CHANGES REQUIRED** :recycle: :recycle: 

**Bright, you've done a good job with your tests. They are passing and it covers all the public methods in your tic tac toe game**. :ok_hand: 

**Great job** :+1: :+1: 

**However, we need to see some mandatory changes effected in order to make this even better and approve the project.** :hammer: :hammer_and_pick: 

**Please see my inline comments as well.**

- [ ] **I can see that you've implemented some negative test cases (e.g move_validate) which is good but some methods still require negative test cases(e.g game_name, sign_validate e.t.c).**
- [ ] **You should have one "it" block for each expected behaviour. Do not test multiple aspect in one "it" block. Please see my inline comments.**

**Implement these changes and you'll be good to go.** :+1: :ok_hand: 

**If you require further clarification, please hit me up on slack or mention me @george-shammar in the comments here.**

**Happy Coding**!!! :computer: :computer_mouse: :+1:

File: spec/logic_spec.rb [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      my_game = Player.new
+      expect(my_game.game_name).to eql('Welcome to TIC TAC TOE game')
+    end
+  end
+
+  describe '#sign_validate' do
+    it 'returns boolean state of players' do
+      sign_validated = Player.new
+      expect(sign_validated.sign_validate('x')).to eql(true)
+      expect(sign_validated.sign_validate('o')).to eql(true)
```
Comment (george-shammar) [inline spec/logic_spec.rb:19]: - [ ] You should have one "it" block per test expectation. Here, you're using one "it" block for multiple test expectations.

File: spec/logic_spec.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    end
+  end
+
+  describe '#move_validate' do
+    it 'returns boolean state of player\'s move' do
+      validate_move = Player.new
+      expect(validate_move.move_validate(1)).to eql(true)
+      expect(validate_move.move_validate(11)).to eql(false)
+      expect(validate_move.move_validate(9)).to equal(true)
+    end
```
Comment (george-shammar) [inline spec/logic_spec.rb:29]: [ ] You should have one "it" block per test expectation. Here, you're using one "it" block for multiple test expectations.

File: spec/logic_spec.rb [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  end
+
+  describe '#index_taken?' do
+    it 'returns the boolean state of board array\'s position' do
+      state = Player.new
+      expect(state.index_taken?(1, [1])).to eql(true)
+      expect(state.index_taken?(8, [7])).to eql(true)
+      expect(state.index_taken?(1, ['x'])).to eql(true) # can't figure this out
+      expect(state.index_taken?(8, ['o'])).to eql(true) # can't figure this out
```
Comment (george-shammar) [inline spec/logic_spec.rb:38]: [ ] You should have one "it" block per test expectation. Here, you're using one "it" block for multiple test expectations.

File: spec/logic_spec.rb [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  end
+
+  describe '#win' do
+    it 'returns boolean winning state' do
+      won = Player.new
+      expect(won.win(%w[o o o], 'o')).to eql(true)
+      expect(won.win(%w[o o o], 'x')).to eql(false)
+      expect(won.win(%w[o x o], 'x')).to eql(false)
+      expect(won.win(%w[x o o], 'o')).to eql(false)
+      expect(won.win(%w[x x x], 'x')).to eql(true)
```
Comment (george-shammar) [inline spec/logic_spec.rb:49]: [ ] You should have one "it" block per test expectation. Here, you're using one "it" block for multiple test expectations.

**Review by george-shammar — CHANGES_REQUESTED** [review]
Please see the inline comments. If you need help, please reach out to me.

**Review by Elbie-em — APPROVED** [review]
## STATUS: APPROVED 🟢 

Hello @brytebee, congratulations on applying all required changes and I am therefore approving this project.

## [OPTIONAL TO IMPROVE]
As this is an OOP project in general, take note of the following:

- [ ] Encapsulation: It would be nice to create different classes responsible for handling differentent tasks in your project. This will make your project stand out and look professional.

### *Keep up the good work and all the best in your next project*

> Elbie Moonga (Part-time TSE)

> Github (@Elbie-em)
> Slack (Elbie Moonga)
> Twitter ([@ElbieEm](https://twitter.com/ElbieEm))
> LinkedIn([Elbie Moonga](https://www.linkedin.com/in/elbiemoonga/))

File: spec/logic_spec.rb (hunk 1 above)
Comment (brytebee) [reply spec/logic_spec.rb]: @george-shammar you mentioned that `spec/logic_spec.rb is outdated`. Please what is its current form?

Comment (george-shammar) [reply spec/logic_spec.rb]: That's an auto message from github

**Pull request merged**